client/src/setup/setupIndexingHooks.ts

```diff
@@ -104,38 +80,3 @@ function setupIndexingLanguageStatusItem(): LanguageStatusItem {
 
   return statusItem;
 }
-
-/**
- * If the doc is open, trigger a noop change on the server to start validation.
- */
-function triggerValidationForOpenDoc(client: LanguageClient) {
-  workspace.textDocuments.forEach((doc) => {
-    // Only trigger files that belong to the project whose worker is ready
-    notifyOfNoopChange(client, doc);
-  });
-}
-
-/**
- * Sends a no-op change notification to the server, this allows the
- * triggering of validation.
- * @param client the language client
- * @param textDoc the open text file to trigger validation on
- */
-function notifyOfNoopChange(client: LanguageClient, textDoc: TextDocument) {
```

Comment:
When I test locally, we get diagnostics on the open editor when indexing closes. Given that we are not sending a noop change, is this what is happening?:

* lsp starts, initialization begins
* editor window with solc error is opened
* document open/change events are queued until initialization ends
* initializtion end, and the events trigger are played, and validation is run

Or is the mechanism something different?

Reply:
Exactly, after the initialize request ends (server sends back a result), other requests can be sent to the server. At that point the client sends `textDocument/didOpen`